parts/k8s/cloud-init/masternodecustomdata.yml

```diff
@@ -68,6 +68,47 @@ write_files:
     {{WrapAsVariable "provisionConfigsCustomCloud"}}
 {{end}}
 
+{{- if HasKubeReservedCgroup .MasterProfile.KubernetesConfig}}
+- path: /etc/systemd/system/{{- GetKubeReservedCgroup .MasterProfile.KubernetesConfig -}}.slice
+  permissions: "0644"
+  owner: root
+  content: |
+    [Unit]
+    Description=Limited resources slice for Kubernetes services
+    Documentation=man:systemd.special(7)
+    DefaultDependencies=no
+    Before=slices.target
+    Requires=-.slice
+    After=-.slice
+    #EOF
```

Comment:
This `#EOF` sentinel chars pattern is meant to work in concert w/ the `wait_for_file` func in the CSE bootstrap scripts. If the following is true:

- in the event that CSE runs prior to cloud-init being finished paving the filesystem, these missing files will prevent CSE from completing successfully/deterministically

...then, we should keep the `#EOF` and add the appropriate `wait_for_file` invocations against these new files (probably inside the `ensureKubelet` func). Unfortunately, becase the CSE files themselves do not operate against a per-pool or per-master context, we have to pass that context via env vars.

Either that, or we can generalize this approach and pick a "default " cgroup so that we *always* deliver this new cgroup implementation flavor, and allow the value of `kubeReservedCgroup` to be user-configured (we could apply that per-node/per-master user configuration in cloud-init, and skip the CSE env var boilerplate thing).

Hope that makes sense!

Reply:
I'm going to always deliver the cgroup and default it to kubereserved, but not run anything in it. That's effectively a no-op unless users manually move stuff into the slice.

Reply:
Is it just me, or should https://github.com/Azure/aks-engine/blob/a43a9c4fcc20cee39d0c7cce84af97734d10686f/parts/k8s/cloud-init/artifacts/cse_config.sh#L317 wait for the file to exist?

Reply:
Yeah, I'm not sure what's going on there...

Reply:
meh, this got a little tricky since the name of the file determines the slice. That doesn't work for the current setup with the system slice, since it already exists (and not via this filepath). I don't really think there's any harm forcing users to pick a slice, but I'm also totally fine with doing this clusterwide and evolving it if we need. That's what I stuck with for now.